backend/server.js

```diff
@@ -23,7 +64,41 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+//Registration endpoint where name, email and password can be assigned to a User in the database 
+app.post('/users', async (req, res) => {
+  try {
+    const { name, email, password } = req.body
+    //TO NOT STORE PLAIN TEXT PASSWORDS
+    const user = new User({ name, email, password: bcrypt.hashSync(password) })
+    const saved = await
+      // const saved = await user.save() res.status(201).json(saved)
+      user.save()
+    res.status(201).json({ id: user._id, accessToken: user.accessToken, message: "✨Created user ✨" })
+  } catch (err) {
+    res.status(400).json({ message: 'Could not create User', errors: err.errors })
+  }
+})
+
+//Secret endpoint
+//Applies the middleware-function above that checks authentication
+app.get('/secrets', authenticateUser)
+app.get('/secrets', (req, res) => {
+  res.json({ message: '"And above all, watch with glittering eyes the whole world around you because the greatest secrets are always hidden in the most unlikely places. Those who don´t believe in magic will never find it."' })
+})
+
+//Logging in endpoint
+app.post('/sessions', async (req, res) => {
+  const user = await User.findOne({ email: req.body.email })
+  //cleartext password from request compared to hashed password in database
+  if (user && bcrypt.compareSync(req.body.password, user.password)) {
+    res.json({ userId: user._id, accessToken: user.accessToken })
+  }
+  else {
+    res.status(403).json({ notFound: true, message: 'The user was not found or entered password is wrong' })
```

Comment:
This is good.  Usually a vague error message during login is good practice so that a potentially malicious user cannot gain information based on message like "The password is wrong (but the user exists)"

Reply:
Regarding your issue with the error messages.  It seems like your '/sessions' endpoint is returning a 200, regardless of what happens on the backend.  Is it possible the latest code is not deployed?
<img width="1367" alt="image" src="https://user-images.githubusercontent.com/807352/75824909-a7d51200-5da4-11ea-80f6-c23b7509037e.png">
**NOTICE**: in the bottom left 200 is returned from `/sessions` but 403 should be.

I tried your code locally and it works as intended 👍